.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,150 @@
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+
+    name: Run Regressions Tests with Python Version ${{ matrix.python-version }}
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Create the logs directory
+        run: mkdir -p logs
+
+      - name: Set filename for the server's output logs
+        run: echo "SERVER_LOGS_FILE=logs/server_logs_${{ matrix.python-version }}" >> $GITHUB_ENV
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Set up GOPATH variable
+        run: echo "GOPATH=$(echo $HOME)" >> $GITHUB_ENV
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.3
+
+      - name: Install aqueduct-ml
+        run: pip3 install aqueduct-ml
+
+      - name: Start the server
+        run: (aqueduct start > $SERVER_LOGS_FILE 2>&1 &)
+
+      - name: Install Packages needed by the notebooks
+        run: pip3 install sklearn transformers torch
+
+      - name: Wait for server
+        timeout-minutes: 1
+        run: while ! echo exit | nc localhost 8080; do sleep 1; done
+      
+      - name: Fetch the API key
+        run: echo "API_KEY=$(aqueduct apikey)" >> $GITHUB_ENV
+
+      - name: Run Sentiment Notebook on previous release
+        working-directory: examples
+        timeout-minutes: 10
+        run: python3 run_notebook.py --path "sentiment_analysis/Sentiment Model.ipynb" --server_address localhost:8080
+
+      - name: Install playwright
+        run: npm install @playwright/test && npx playwright install && npx playwright install-deps
+
+      - name: Take screenshot of old workflow run 1
+        working-directory: regression_tests
+        timeout-minutes: 5
+        run: imgname=old-flow-run-1 npx playwright test tests/screenshot.test.ts
+
+      - name: Trigger new workflow run 
+        working-directory: regression_tests
+        timeout-minutes: 5
+        run: npx playwright test tests/trigger.test.ts --project=chromium
+
+      - name: Take screenshot of old workflow run 2
+        working-directory: regression_tests
+        timeout-minutes: 5
+        run: imgname=old-flow-run-2 npx playwright test tests/screenshot.test.ts
+
+      - name: Take checkpoint of workflow
+        working-directory: regression_tests
+        timeout-minutes: 5
+        run: sleep 5 && python3 tests/workflow_comparer.py --server_address localhost:8080 --checkpoint=create --path=checkpoint
+      
+      # Grabs the pid of the process bound to port 8080 and kills it
+      - name: Kill the server
+        run: kill -9 $(lsof -nP -iTCP -sTCP:LISTEN | grep 8080 | awk '{print $2}')
+      
+      # install_local.py requires ~/.aqueduct to exist.
+      - name: Update aqueduct with latest code
+        run: python3 scripts/install_local.py --gobinary --sdk --executor
+
+      - name: Start the server again
+        run: (aqueduct start > $SERVER_LOGS_FILE 2>&1 &)
```

Comment:
we should probably dump the server logs of the second run into a different log file?